patches/net/minecraft/client/particle/ParticleRenderType.java.patch

```diff
@@ -0,0 +1,11 @@
+--- a/net/minecraft/client/particle/ParticleRenderType.java
++++ b/net/minecraft/client/particle/ParticleRenderType.java
+@@ -126,4 +_,8 @@
+     void begin(BufferBuilder p_107436_, TextureManager p_107437_);
+ 
+     void end(Tesselator p_107438_);
++
++    default boolean isTranslucent() {
++        return this == PARTICLE_SHEET_TRANSLUCENT;
```

Comment:
I'm not sure only checking for `PARTICLE_SHEET_TRANSLUCENT` is sufficient as `TERRAIN_SHEET` (used for sprinting and block break particles) also enables blending.
Overall it might make more sense to default to `true` and then override to return `false` in `PARTICLE_SHEET_OPAQUE`, `PARTICLE_SHEET_LIT` and `CUSTOM` to retain existing behavior for modded particles and allow them to opt-in to the fixed behavior if they need it to prevent accidentally pushing translucent particles into the non-translucent pass.

Reply:
Sure; `TERRAIN_SHEET` doesn't appear to be used by most vanilla particles anyway. Mods that use the terrain atlas to render non-translucent particles will have to make their own copy of `TERRAIN_SHEET` that returns `false`, but that shouldn't be a big deal.